components/mqtt/publish.go

```diff
@@ -157,21 +157,34 @@ func (s *Server) PublishBlockMetadata(metadata *inx.BlockMetadata) {
 	}
 }
 
-func payloadForOutput(api iotago.API, iotaOutput iotago.Output) *outputPayload {
-	rawOutputJSON, err := api.JSONEncode(iotaOutput)
-	if err != nil {
-		return nil
-	}
-
-	rawRawOutputJSON := json.RawMessage(rawOutputJSON)
+func payloadForOutput(ledgerIndex iotago.SlotIndex, output *inx.LedgerOutput, iotaOutputJSON []byte) *outputPayload {
+	outputID := output.GetOutputId().Unwrap()
+	rawRawOutputJSON := json.RawMessage(iotaOutputJSON)
```

Comment:
Why don't we serialize the `iotaOutput` inside of this function like it is done in stardust?

Reply:
Then we need to pass API to `payloadForOutput` to encode it correctly. Since only the JSON encoded `iotago.Output` is needed, I thought we could just do it in the upper level.

Reply:
Mhh I would pass it down, as it is done right now in stardust.
Otherwise you have 2 places where you need to do the json encoding of the output. This way you have only one.